js/index.js

```diff
@@ -44,26 +44,94 @@ function renderPepperoni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(oneMushroom => {
+    if (state.mushrooms) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(oneGreenPepper => {
+    if (state.greenPeppers) {
+      oneGreenPepper.style.visibility = 'visible';
+    } else {
+      oneGreenPepper.style.visibility = 'hidden';
+    }
+  })
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  if (state.whiteSauce) {
+    document.querySelector('.sauce').classList.add('sauce-white');
+  }
+  else {
+    document.querySelector('.sauce').classList.remove('sauce-white');
+  }
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  if (state.glutenFreeCrust) {
+    document.querySelector('.crust').classList.add('crust-gluten-free');
+  } else {
+    document.querySelector('.crust').classList.remove('crust-gluten-free');
+  }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  if (!state.pepperoni) {
+    document.querySelector('.btn.btn-pepperoni').classList.remove('active');
+  } else {
+    document.querySelector('.btn.btn-pepperoni').classList.add('active');
+  }
+  if (!state.mushrooms) {
+    document.querySelector('.btn.btn-mushrooms').classList.remove('active');
+  } else {
+    document.querySelector('.btn.btn-mushrooms').classList.add('active');
+  }
+  if (!state.greenPeppers) {
+    document.querySelector('.btn.btn-green-peppers').classList.remove('active');
+  } else {
+    document.querySelector('.btn.btn-green-peppers').classList.add('active');
+  }
+  if (state.whiteSauce) {
+    document.querySelector('.btn.btn-sauce').classList.add('active');
+  } else {
+    document.querySelector('.btn.btn-sauce').classList.remove('active')
+  }
+  if (state.glutenFreeCrust) {
+    document.querySelector('.btn.btn-crust').classList.add('active');
+  } else {
+    document.querySelector('.btn.btn-crust').classList.remove('active')
+  }
 }
```

Comment:
nice approach!
As an example, there is also this approach that is shorter and cleaner.

`function renderButtons() { var ingArr = Object.keys(state); document.querySelectorAll(".btn").forEach(function(item, index) { if (state[ingArr[index]]) { item.classList.add("active"); } else { item.classList.remove("active"); } }); }`